src/clients/AcrossConfigStoreClient.ts

```diff
@@ -56,14 +66,20 @@ export class AcrossConfigStoreClient {
     readonly logger: winston.Logger,
     readonly configStore: Contract,
     readonly eventSearchConfig: MakeOptional<EventSearchConfig, "toBlock"> = { fromBlock: 0, maxBlockLookBack: 0 },
-    protected readonly configOverride: {
-      enabledChainIds: number[];
-      defaultConfigStoreVersion: number;
-      configStoreVersion: number;
-    }
+    {
+      enabledChainIds = CHAIN_ID_LIST_INDICES,
+      configStoreVersion = CONFIG_STORE_VERSION,
+      defaultConfigStoreVersion = DEFAULT_CONFIG_STORE_VERSION,
+    }: { enabledChainIds?: number[]; configStoreVersion?: number; defaultConfigStoreVersion?: number } = {}
```

Comment:
The syntax here is pretty obtuse, but it permits the caller to independently set any of the set of { enabledChainIds, configStoreVersion, defaultConfigStoreVersion } as they please.

Reply:
My only qualm with this is that it permits a set of opinionated values to this SDK. I'm not sure we should set constants inside the SDK itself but add additional documentation that mentions how to use it best. 

I'm happy to approve this pending @mrice32 sign-off.

Reply:
It's necessary for the code instantiating the class to specify which config version that are using, so I don't think we can do away with that. I am less confident about overriding the default config store value though - we have strong comments saying that it should only ever be set to 0, so I wonder if it needs to be externally configurable, or whether it can just be internal/private to the ConfigStoreClient.

Reply:
@james-a-morris Have updated after your comments. The user is now required to be explicit about their maximum supported config store version and supported chainIds. I've reinstated the corresponding changes over in the draft PR for relayer-v2: https://github.com/across-protocol/relayer-v2/pull/703.